src/mito2/src/sst/index/fulltext_index/creator.rs

```diff
@@ -0,0 +1,293 @@
+    FulltextOptionsSnafu, FulltextPushTextSnafu, OperateAbortedIndexSnafu, PuffinAddBlobSnafu,
+    Result,
+};
+use crate::read::Batch;
+use crate::sst::file::FileId;
+use crate::sst::index::fulltext_index::INDEX_BLOB_TYPE;
+use crate::sst::index::intermediate::IntermediateManager;
+use crate::sst::index::puffin_manager::SstPuffinWriter;
+use crate::sst::index::statistics::{ByteCount, RowCount, Statistics};
+
+/// `SstIndexCreator` is responsible for creating fulltext indexes for SST files.
+pub struct SstIndexCreator {
+    /// Creators for each column.
+    creators: HashMap<ColumnId, SingleCreator>,
+    /// Whether the index creation was aborted.
+    aborted: bool,
+    /// Statistics of index creation.
+    stats: Statistics,
+}
+
+impl SstIndexCreator {
+    /// Creates a new `SstIndexCreator`.
+    pub async fn new(
+        region_id: &RegionId,
+        sst_file_id: &FileId,
+        intermediate_manager: &IntermediateManager,
+        metadata: &RegionMetadataRef,
+        compress: bool,
+        mem_limit: usize,
+    ) -> Result<Self> {
+        let mut creators = HashMap::new();
+
+        for column in &metadata.column_metadatas {
+            let options =
+                column
+                    .column_schema
+                    .fulltext_options()
+                    .context(FulltextOptionsSnafu {
+                        column_name: &column.column_schema.name,
+                    })?;
+
+            // Relax the type constraint here as many types can be casted to string.
+
+            let options = match options {
+                Some(options) if options.enable => options,
+                _ => continue,
+            };
+
+            let column_id = column.column_id;
+            let intm_path = intermediate_manager.fulltext_path(region_id, sst_file_id, column_id);
+
+            let config = Config {
+                analyzer: match options.analyzer {
+                    FulltextAnalyzer::English => Analyzer::English,
+                    FulltextAnalyzer::Chinese => Analyzer::Chinese,
+                },
+                case_sensitive: options.case_sensitive,
+            };
+
+            let creator = TantivyFulltextIndexCreator::new(&intm_path, config, mem_limit)
+                .await
+                .context(CreateFulltextCreatorSnafu)?;
+
+            creators.insert(
+                column_id,
+                SingleCreator {
+                    column_id,
+                    inner: Box::new(creator),
+                    intm_path,
+                    compress,
+                },
+            );
+        }
+
+        Ok(Self {
+            creators,
+            aborted: false,
+            stats: Statistics::default(),
+        })
+    }
+
+    /// Updates the index with the given batch.
+    pub async fn update(&mut self, batch: &Batch) -> Result<()> {
+        ensure!(!self.aborted, OperateAbortedIndexSnafu);
```

Comment:
What about adding some metrics? About the timing and data size etc.

Reply:
In do_update, `stats` will record the metrics and flush when dropped. However, you reminded me that it is possible to distinguish between the metrics of the inverted index and the full-text index.